pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -452,7 +452,7 @@ func (o *MustGatherOptions) logTimestamp() error {
 func (o *MustGatherOptions) copyFilesFromPod(pod *corev1.Pod) error {
 	streams := o.IOStreams
 	streams.Out = newPrefixWriter(streams.Out, fmt.Sprintf("[%s] OUT", pod.Name))
-	destDir := path.Join(o.DestDir, regexp.MustCompile("[^A-Za-z0-9]+").ReplaceAllString(pod.Status.ContainerStatuses[0].ImageID, "-"))
+	destDir := path.Join(o.DestDir, regexp.MustCompile("[^A-Za-z0-9]+").ReplaceAllString(pod.Status.ContainerStatuses[0].ImageID+pod.Spec.NodeName, "-"))
```

Comment:
Can we make this optional when you actually specify node selector such that the structure will look like:
- destDir/
    - node1/image-name
    - node2/image-name
instead of appending those. It's already pretty lengthy name and adding node name at the end won't be helpful :/

Reply:
Done, ptal